plugin/edge/edge.go

```diff
@@ -1,106 +1,238 @@
+// Adapted from https://github.com/coredns/coredns/blob/master/plugin/forward/forward.go
+
 package edge
 
 import (
 	"crypto/tls"
-	"encoding/json"
-	"errors"
-	"fmt"
 	"io"
 	"net"
-	"regexp"
-	"strconv"
+	"net/http"
 	"time"
 
 	"github.com/coredns/coredns/plugin"
 	"github.com/coredns/coredns/request"
-	"wwwin-github.cisco.com/edge/optikon-dns/plugin/central"
-
 	"github.com/miekg/dns"
 	ot "github.com/opentracing/opentracing-go"
 	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	pluginName              = "edge"
+	defaultSvcReadInterval  = 2 * time.Second
+	defaultSvcPushInterval  = 3 * time.Second
+	defaultExpire           = 10 * time.Second
+	defaultMaxUpstreamFails = 2
+	maxUpstreams            = 15
```

Comment:
Just curious, why is max upstreams `15`? does this number come from anywhere?

Reply:
This was the hardcoded max in the origin `forward` plugin. Not sure why, but it seemed like a reasonable choice. You can always change it if you want.

Reply:
np, just curious